lib/use_cases/save_customer_details.rb

```diff
@@ -6,9 +6,93 @@ def initialize(customer_gateway:)
   end
 
   def execute(customer_details:)
+    @customer_details = customer_details
+    errors = check_for_missing_fields
+    errors.push(check_phone_numbers)
+    errors.push(check_postcodes)
+    errors.push(check_emails)
+    errors.flatten!
+
```

Comment:
Maybe extract all the validation into one validation method, returning the errors array